src/plugins/data/README.md

```diff
@@ -1,9 +1,137 @@
 # data
 
-`data` plugin provides common data access services.
+The `data` plugin provides common data access services for solutions and application developers.
 
-- `expressions` &mdash; run pipeline functions and render results.
-- `filter`
-- `index_patterns`
-- `query`
-- `search`: Elasticsearch API service and strategies
\ No newline at end of file
+## Autocomplete
+
+The autocomplete service can provide suggestions for field names and field values.
+
+It is pre-wired into the `TopNavMenu` component, but can be used independently.
+
+### Fetch Query Suggestions
+
+The `getQuerySuggestions` helps to construct a query. 
+KQL suggestion functions are registered in x-pack, so this API won't return results in OSS.
+
+```.ts
+
+    // `inputValue` is the user input
+    const querySuggestions = await autocomplete.getQuerySuggestions({
+        language: 'kuery',
+        indexPatterns: [indexPattern],
+        query: inputValue,
+    });
+
+```
+
+### Fetch Value Suggestions
+
+The `getValueSuggestions` returns suggestions for possible field values.
+This is helpful when you want to provide a user with options to choose from, for example when constructing a filter. 
+
+```.ts
+
+    // `inputValue` is the user input
+    const valueSuggestions = await autocomplete.getValueSuggestions({
+      indexPattern,
+      field,
+      query: inputValue,
+    });
+
+```
+
+## Field Formats
+
+Coming soon.
+
+## Index Patterns
+
+Coming soon.
+
+## Query
+
+The query service is responsible for managing the configuration of a search query (`QueryState`): filters, timerange and query string, as well as other settings such as the auto refresh behavior and saved queries.
+
+It contains sub-services for each of those configurations:
+ - `data.query.filterManager` - Manages the `filters` component of a `QueryState`. The global filter state (filters that are persisted between applications) are owned by this service.
+ - `data.query.timefilter` - Responsible for the time range filter as well as the auto refresh behavior settings.
+ - `data.query.queryString` - Responsible for the query string and query language settings.
+ - `data.query.savedQueries` - This sub-service is responsible for persisting a `QueryState` into a `SavedObject`, so it can be restored and re-used by other applications.
```

Comment:
```suggestion
 - `data.query.savedQueries` - Responsible for persisting a `QueryState` into a `SavedObject`, so it can be restored and re-used by other applications.
```